lamkey_nick_ndvi.py

```diff
@@ -0,0 +1,723 @@
+        A geopandas dataframe to be used to crop the raster data using rasterio mask().
+    pixel_qa_path: string
+        A path to a pixel qa tif file.
+    vals: list
+        A list of values needed to create the cloud mask
+
+
+    Returns
+    -----------
+    ndvi_crop : Xarray Dataset
+        a cropped and masked xarray object containing NDVI values
+    """
+
+    crop_json = crop_bound.geometry
+
+    # Open and clip qa layer
+    pixel_qa = rxr.open_rasterio(pixel_qa_path, masked=True).rio.clip(crop_json,
+                                                                      from_disk=True).squeeze()
+
+    # Calculate NDVI
+    ndvi_xr = (all_bands[1]-all_bands[0]) / (all_bands[1]+all_bands[0])
+
+    # Apply cloud mask to NDVI
+    ndvi_mask = ndvi_xr.where(~pixel_qa.isin(vals))
+
+    return ndvi_mask
+
+
+# ## Processing multiple scenes from one site
+# ### Where did the data come from?
+# * The Landsat data originally comes from the USGS Landsat 8. Landsat data can be downloaded via earthexplorer. The data for this workflow can be downloaded using earthpy: data = et.data.get_data('ndvi-automation'). Scenes from Jan 2017 to December 2017 were analyzed for the Harvard forest (HARV) in the Eastern United States.
+# * The numbers for vals, used to masked clouds from the scenes, are the USGS website:https://www.usgs.gov/media/files/landsat-8-collection-1-land-surface-reflectance-code-product-guide
+# 
+# ### What's the workflow?
+# * Below a loop was written to process multiple landsat scenes for one site. A master for loop utilizes two functions that process the data so that mean NDVI can be calculated for each scene. The first function opens the data, crops it to the site extents and filters for a range of values that eliminate negative values. The second for loop uses the pixel_qa layer that comes with Landsat imagery to mask clouds that may be present in any scenes. This is to ensure the reflectance values from the clouds are not included in the final calculation. The function crops the pixel_qa layer, masks each band and calculates the NDVI. Within the master for loop, site name, dates associated with each site and mean NDVI are captured into lists and then added to a dataframe. 
+# 
+# ### How was the code optimized?
+# * The xarrays being worked on in the for loop were stored in a list within the for loop so that only two bands were stored in the list per iteration. This made it easy to run the mask_crop_ndvi function on both bands while keeping the data in order. Pyproj was used to speed up processing times.  
+
+# In[6]:
+
+
+# Create dataframe of mean NDVI in this cell using the functions created above
+# Important: to use the ungraded tests below as a sanity check,
+# name your columns: mean_ndvi and site
+# Call the dataframe at the end of the cell so the tests run on it!
+# Be sure that the date column is an index of type date
+# HINT: the time series lessons may help you remember how to do this!
+
+# Paths to data
```

Comment:
@nlamkey paths throughout the code appear to be set correctly for extensibility across all os's